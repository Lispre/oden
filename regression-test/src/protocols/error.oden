package protocols/error/main

import strconv

// protocol error (builtin)

impl error(string) {
  Error(s) = "Error: " ++ s
}

impl error(int) {
  Error(code) = "E" ++ strconv.Itoa(code)
}

impl error({ error: string }) {
  Error(e) = e.error ++ " (no message)"
}

impl error({ error: string, message: string }) {
  Error(e) = e.error ++ ": " ++ e.message
}

main() = {
  println(error::Error(192))
  println(error::Error({
    error = "Unknown failure"
  }))
  println(error::Error({
    error = "Stuff broke",
    message = "But protocols seems to be working still, right?"
  }))
  println(error::Error(error::Error(192)))
}
