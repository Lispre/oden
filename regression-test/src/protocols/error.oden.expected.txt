E192
Unknown failure (no message)
Stuff broke: But protocols seems to be working still, right?
Error: E192
